RUFAS/routines/field/field/field.py

```
@@ -39,7 +40,8 @@ def __init__(self, field_data: Optional[FieldData] = None, soil: Optional[Soil]
                  fertilizer_events: Optional[List[FertilizerEvent]] = None,
                  fertilizer_mixes: Optional[Dict[str, Dict[str, float]]] = None,
                  manure_events: Optional[List[ManureEvent]] = None,
-                 manure_manager: Optional[ManureManager] = None):
+                 manure_manager: Optional[ManureManager] = None,
+                 feed_manager: Optional[FeedManager] = None):
```

Why is it optional? What happens if it is None?

---

If it is None, the field will instantiate its own `FeedManager`.

---

Would you like to document it?

---

Yes - I am not sure if this is the best approach to handling a Field instantiated without a FeedManager though. Is this something we could reasonably expect to happen in RuFaS? If not, then wouldn't it be better to raise an error and fail initialization if no FeedManager is passed?

---

I think raising an error is a better approach.

---

I will do this and also apply the same logic to the `ManureManager`

Update: changing this code is important and is not particularly hard to do. The issue lies in the unit tests - there are a lot of initializations that would need to change (and I mean a lot - it would significantly increase the size and scope of this PR). Would an acceptable alternative be to raise an error to OM when a Field attempts to initialize without a FeedManager, and we can revisit the initialization routine for Field (and other modules in C&S which have the similar designs) during the Fixathon?

---

Let's:
1. make an issue about this, we may or may not do it during the fixathon
2. log an error to OM when it happens.

---

Written up in #1104.